pkg/server/api/bundle/v1/service_test.go

```diff
@@ -358,9 +339,153 @@ func createBundle(t *testing.T, test *serviceTest, td string) *common.Bundle {
 	return b
 }
 
-func assertBundleWithMask(t *testing.T, expected *common.Bundle, actual *types.Bundle, m *types.BundleMask) {
+func TestBatchCreateFederatedBundle(t *testing.T) {
+	test := setupServiceTest(t)
+	defer test.Cleanup()
+
+	for _, tt := range []struct {
+		name            string
+		bundlesToCreate []*types.Bundle
+		outputMask      *types.BundleMask
+		expectedResults []*bundlepb.BatchCreateFederatedBundleResponse_Result
+		expectedLogMsgs []spiretest.LogEntry
+		dsError         error
+	}{
+		{
```

Comment:
maybe you can add a test case where you create more than one bundle and someone fails and another not, and verify that we always keep order, and expected logs+results

Reply:
another test case to add is a mask with all false, what happens in that case? and when it is nil?

Reply:
another test case when bundles is empty?

Reply:
> maybe you can add a test case where you create more than one bundle and someone fails and another not, and verify that we always keep order, and expected logs+results

This scenario is covered by "Create fails if bundle already exists"

> another test case to add is a mask with all false, what happens in that case? and when it is nil?

Will do